File: LeetCode/1029_Two_City_Scheduling.js
/*
1029. Two City Scheduling

Q: https://leetcode.com/problems/two-city-scheduling/
*/

/**
 * @param {number[][]} costs
 * @return {number}
 */
function compareHelper(a, b) {
  if (a[1] - a[0] >= b[1] - b[0]) return 1;
  return -1;
}

function twoCitySchedCost(costs) {
  let res = 0;
  let len = costs.length;
  costs.sort(compareHelper);
  for (let i = 0; i < len; ++i) {
    if (i < len / 2) res += costs[i][1];
    else res += costs[i][0];
  }
  return res;
}

File: LeetCode/1029_Two_City_Scheduling.py
"""
1029. Two City Scheduling

Q: https://leetcode.com/problems/two-city-scheduling/
"""


class Solution:
    def twoCitySchedCost(self, costs):
        res = 0
        l = int(len(costs)/2)
        costs.sort(key=lambda c: c[1] - c[0])
        for idx, costPair in enumerate(costs):
            if idx >= l:
                res += costPair[0]
            else:
                res += costPair[1]
        return res
